main.nf

```diff
@@ -312,135 +468,210 @@ process get_software_versions {
+    process gunzip_transcript_fasta {
+        tag "$gz"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_transcriptome" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file gz from transcript_fasta_gz
+
+        output:
+        file "${gz.baseName}" into ch_fasta_for_salmon_index
+
+        script:
+        """
+        gunzip --verbose --stdout --force ${gz} > ${gz.baseName}
+        """
+    }
+  }
+  if (params.bed12){
+    process gunzip_bed12 {
+        tag "$gz"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_genome" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file gz from bed12_gz
+
+        output:
+        file "${gz.baseName}" into bed_rseqc
+
+        script:
+        """
+        gunzip --verbose --stdout --force ${gz} > ${gz.baseName}
+        """
+    }
+  }
+  if (!params.skipAlignment && params.star_index && params.aligner == "star"){
+    process gunzip_star_index {
+        tag "$gz"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_genome/star" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file gz from star_index_gz
+
+        output:
+        file "${gz.simpleName}" into star_index
+
+        script:
+        // Use tar as the star indices are a folder, not a file
+        """
+        tar -xzvf ${gz}
+        """
+    }
+  }
+  if (!params.skipAlignment && params.hisat2_index && params.aligner == 'hisat2'){
+    process gunzip_hisat_index {
+        tag "$gz"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_genome/hisat2" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file gz from hs2_indices_gz
+
+        output:
+        file "*.ht2*" into hs2_indices
+
+        script:
+        // Use tar as the hisat2 indices are a folder, not a file
+        """
+        tar -xzvf ${gz}
+        """
+    }
+  }
+  if (params.salmon_index && params.pseudo_aligner == 'salmon'){
+    process gunzip_salmon_index {
+        tag "$gz"
+        publishDir path: { params.saveReference ? "${params.outdir}/reference_transcriptome/hisat2" : params.outdir },
+                   saveAs: { params.saveReference ? it : null }, mode: 'copy'
+
+        input:
+        file gz from salmon_index_gz
+
+        output:
+        file "${gz.simpleName}" into salmon_index
+
+        script:
+        // Use tar as the hisat2 indices are a folder, not a file
+        """
+        tar -xzvf ${gz}
+        """
+    }
+  }
 }
+
 /*
  * PREPROCESSING - Convert GFF3 to GTF
  */
-if(params.gff){
-  process convertGFFtoGTF {
-      tag "$gff"
+if(params.gff && !params.gtf){
```

Comment:
I do prefer using when statements within a process, but that's definitively my personal opinion

Reply:
I think when DSLv2 is coming out much of this is going to be re-done using when statements. `when` also requires channel mixing and other specialties such as setting Channels to empty, to not cause issues with re-using channels, for example, making it quite hard to get it right...

I also like `when` more 👍